sim/midas/src/main/scala/midas/widgets/AssertBridge.scala

```diff
@@ -18,8 +18,15 @@ class AssertBridgeRecord(assertPortName: String, resetPortName: String, numAsser
   override def cloneType = new AssertBridgeRecord(assertPortName, resetPortName, numAsserts).asInstanceOf[this.type]
 }
 
-class AssertBridgeModule(assertPortName: String, resetPortName: String, assertMessages: Seq[String])(implicit p: Parameters)
+case class AssertBridgeParameters(assertPortName: String, resetPortName: String, assertMessages: Seq[String])
+
+class AssertBridgeModule(key: AssertBridgeParameters)(implicit p: Parameters)
     extends BridgeModule[HostPortIO[AssertBridgeRecord]]()(p) {
+
+  val assertPortName = key.assertPortName
```

Comment:
This is neither here nor there, but you can destructure the case class with its unapply method like so:
` val AssertBridgeParameters(assertPortName, resetPortName, assertMessages) = key`